git_machete/github.py

```diff
@@ -40,19 +46,19 @@ def fire_github_api_get_request(url: str, token: Optional[str]) -> Any:
     conn: HTTPSConnection = HTTPSConnection(host)
 
     try:
-        conn.request('GET', url, body=None, headers=headers)
+        conn.request(method, url, body=json.dumps(request_body), headers=headers)
 
         response: HTTPResponse = conn.getresponse()
-        body: Any = json.loads(response.read().decode())
+        parsed_response_body: Any = json.loads(response.read().decode())
 
         if 200 <= response.status < 300:
-            return body
+            return parsed_response_body
         else:
-            first_line = fmt(f'GitHub API returned {response.status} HTTP status with error message: `{body.get("message")}`.\n')
+            first_line = fmt(f'GitHub API returned {response.status} HTTP status with error message: `{parsed_response_body.get("message")}`\n')
             if token:
                 raise MacheteException(
                     first_line + fmt(f'Make sure that the token provided in <b>{GITHUB_TOKEN_ENV_VAR}</b> env var is valid '
-                                     f'and allows for access to `GET https://{host}{url}`.'))
+                                     f'and allows for access to {method} https://{host}{url}.'))
```

Comment:
This message sometimes doesn't make sense:

```
➜ 19:41 ~/infra-sandbox bar PR #16 $ git machete hub create-pr 
Creating a PR from bar to master... OK, see https://github.com/PawelLipski/infra-sandbox/pull/17
Setting milestone of PR #17 to 1... OK
Adding PawelLipski as assignee to PR #17... OK
Adding git-machete-bot as reviewers to PR #17...
GitHub API returned 422 HTTP status with error message: Reviews may only be requested from collaborators. One or more of the users or teams you specified is not a collaborator of the PawelLipski/infra-sandbox repository.
Make sure that the token provided in GITHUB_TOKEN env var is valid and allows for access to POST https://api.github.com/repos/PawelLipski/infra-sandbox/pulls/17/requested_reviewers.
```

Reply:
Probably it should only be displayed for 403 (and 401?) status

Reply:
Hmmm okay so is there any special handling of 401/403 status now? 🤔

Reply:
No, there isn't. But is there something we can still do in case of 401/403 Status?

Reply:
I'd put it in a different way: the existing error messages (`Make sure that the token provided ...` and `This repository might be private. Provide a GitHub API token...`) only make sense in case of 401/403 status.

Reply:
In case of a different status &mdash; it's hard to provide a ready response, it might be better to just display the details with http status + response, and even maybe a link to create an issue against git-machete?

Reply:
The message to open an issue will also be included in UnprocessableEntity error, as it's considered as bug.